apps/blockchain/lib/blockchain/contract/create_contract.ex

```diff
@@ -127,24 +127,30 @@ defmodule Blockchain.Contract.CreateContract do
     contract_creation_cost = creation_cost(output)
     insufficient_gas = remaining_gas < contract_creation_cost
 
-    if insufficient_gas && EVM.Configuration.fail_contract_creation_lack_of_gas?(params.config) do
-      {:error, {params.state, 0, SubState.empty()}}
-    else
-      resultant_gas =
-        if insufficient_gas do
-          remaining_gas
-        else
-          remaining_gas - contract_creation_cost
-        end
-
-      resultant_state =
-        if insufficient_gas do
-          state_after_init
-        else
-          Account.put_code(state_after_init, address, output)
-        end
-
-      {:ok, {resultant_state, resultant_gas, accrued_sub_state}}
+    cond do
+      insufficient_gas && EVM.Configuration.fail_contract_creation_lack_of_gas?(params.config) ->
+        {:error, {params.state, 0, SubState.empty()}}
+
+      # EIP170 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-170.md
+      byte_size(output) > 24_577 ->
```

Comment:
What do you think about extracting this condition to a function that is more self-explanatory? 

```elixir
# in the `cond` block 
eip_170_byte_size?(output) -> 

# then further below
def eip_170_byte_size?(output), do: byte_size(output) > 24_577

# or even better, add docs
@doc """
This is implemented because of EIP170
See more at https://github.com/ethereum/EIPs/blob/master/EIPS/eip-170.md
"""
def eip_170_byte_size?(output), do: byte_size(output) > 24_577
```

What do you think?

Reply:
yes, good idea. btw it looks like this change affects previous tests so I'll rewrite it using `EVM.Configuration`